lib/verk/workers_manager.ex

```diff
@@ -99,7 +99,15 @@ defmodule Verk.WorkersManager do
     if workers != 0 do
       case QueueManager.dequeue(state.queue_manager_name, workers) do
         jobs when is_list(jobs) ->
-          for job <- jobs, do: job |> Job.decode! |> start_job(state)
+          Enum.map(jobs, fn job ->
+            try do
+              job |> Job.decode! |> start_job(state)
```

Comment:
I'm fine either by just logging or adding to the dead set. We just need to add to the dead set before `ack` so that we don't have a ghost job:

For example:

* Ack!
* Add dead job (fails) (BOOM)

The job is gone from progress and it wasn't added to the dead set.

If we go to the Dead Set route we could potentially use the original json and add a clause for the job being already a string? so DeadSet.add would have an implementation if a Job is sent and one if a string is sent?